runtime/src/append_vec.rs

```diff
@@ -451,6 +490,47 @@ pub mod tests {
         }
     }
 
+    #[test]
+    #[should_panic(expected = "too small file size 0 for AppendVec")]
```

Comment:
I know those `should_panic`s are rather ugly. I'll brush up once this PR's direction is affirmed.

Reply:
Other uses are fixed by now. Only this will remain as this is testing a panic from `::new()`.